include/aws/common/array_list.h

```diff
@@ -30,107 +30,113 @@ struct aws_array_list {
-    * behavior.
-    */
-    AWS_COMMON_API int aws_array_list_pop_front(struct aws_array_list *list);
-
-    /**
-     * Copies the element at the end of the list if it exists. If list is empty, AWS_ERROR_LIST_EMPTY will be raised.
-     */
-    AWS_COMMON_API int aws_array_list_back(const struct aws_array_list *list, void *val);
-
-    /**
-    * Deletes the element at the end of the list if it exists. If list is empty, AWS_ERROR_LIST_EMPTY will be raised.
-    */
-    AWS_COMMON_API int aws_array_list_pop_back(struct aws_array_list *list);
-
-    /**
-     * Clears all elements in the array and resets length to zero. Size does not change in this operation.
-     */
-    AWS_COMMON_API void aws_array_list_clear(struct aws_array_list *list);
-
-    /**
-     * If in dynamic mode, shrinks the allocated array size to the minimum amount necessary to store its elements.
-     */
-    AWS_COMMON_API int aws_array_list_shrink_to_fit(struct aws_array_list *list);
-
-    /**
-     * Copies the elements from from to to. If to is in static mode, it must at least be the same length as from. Any data
-     * in to will be overwritten in this copy.
-     */ 
-    AWS_COMMON_API int aws_array_list_copy(const struct aws_array_list *from, struct aws_array_list *to);
-
-    /**
-     * Returns the number of elements that can fit in the internal array. If list is initialized in dynamic mode,
-     * the capacity changes over time.
-     */
-    AWS_COMMON_API size_t aws_array_list_capacity(const struct aws_array_list *list);
-
-    /**
-     * Returns the number of elements in the internal array.
-     */
-    AWS_COMMON_API size_t aws_array_list_length(const struct aws_array_list *list); 
-     
-    /**
-     * Copies the memory at index to val. If element does not exist, AWS_ERROR_INVALID_INDEX will be raised.
-     */
-    AWS_COMMON_API int aws_array_list_get_at(const struct aws_array_list *list, void *val, size_t index);
-
-    /**
-     * Copies the memory address of the element at index to *val. If element does not exist, AWS_ERROR_INVALID_INDEX will be raised.
-     */
-    int aws_array_list_get_at_ptr(const struct aws_array_list *list, void **val, size_t index);
-
-    /**
-     * Copies the the memory pointed to by val into the array at index. If in dynamic mode, the size will grow by a factor of two
-     * when the array is full. In static mode, AWS_ERROR_INVALID_INDEX will be raised if the index is past the bounds of the array.
-     */
-    AWS_COMMON_API int aws_array_list_set_at(struct aws_array_list *list, const void *val, size_t index);
-
-    /**
-     * Swap elements at the specified indices.
-     */
-    void aws_array_list_swap(struct aws_array_list *list, size_t a, size_t b);
+/**
+ * Initializes an array list with an array of size initial_item_allocation * item_size. In this mode, the array size
+ * will
```

Comment:
these probably need manual correction.

Reply:
clang-format doesn't realize there's a brace that's #ifdef-ed. 
I think it's ok to let this one slip.